fix: ensure aria-selected attribute is removed from the item clone

web-padawan opened the pull request:
## Description

- Updated `selected` property in `ItemMixin` to use `sync: true` so it triggers observer synchronously
- Added call to `performUpdate()` on the item clone in `vaadin-select` to remove `aria-selected`
- Added missing unit test to ensure `aria-selected` attribute is removed (in Lit version it was not)

Note: there was an error in the "keyboard selection" test apparently caused by the fact that  "append value item" was called too late. Added `await nextUpdate()` to this test helped to remove the error so I think it was a test issue.

```
        at SelectItem.update (node_modules/lit-element/src/lit-element.ts:163:23)
        at SelectItem.performUpdate (node_modules/@lit/reactive-element/src/reactive-element.ts:1440:13)
        at Select.__appendValueItemElement (packages/select/src/vaadin-select-base-mixin.js:467:21)
        at Select.__attachSelectedItem (packages/select/src/vaadin-select-base-mixin.js:430:12)
```

## Type of change

- Bugfix

vursen commented:
**Issue:** The `aria-selected` attribute is still not removed when populating the select with items that have a `label` attribute.

```html
<vaadin-select></vaadin-select>

<script type="module">
  import '@vaadin/select';
  import '@vaadin/list-box';
  import '@vaadin/item';
  import { render, html } from 'lit';

  const select = document.querySelector('vaadin-select');
  select.renderer = (root) => {
    render(
      html`
        <vaadin-list-box>
          <vaadin-item label="Label 0" value="value-0">Option 0</vaadin-item>
          <vaadin-item label="Label 1" value="value-1">Option 1</vaadin-item>
        </vaadin-list-box>
      `,
      root,
    );
  };
</script>
```

vursen commented:
**Issue:** The `aria-selected` attribute is still not removed when the select is populated with items that have a `label` attribute. This actually happens for both Lit and Polymer.

```html
<vaadin-select></vaadin-select>

<script type="module">
  import '@vaadin/select';
  import '@vaadin/list-box';
  import '@vaadin/item';
  import { render, html } from 'lit';

  const select = document.querySelector('vaadin-select');
  select.renderer = (root) => {
    render(
      html`
        <vaadin-list-box>
          <vaadin-item label="Label 0" value="value-0">Option 0</vaadin-item>
          <vaadin-item label="Label 1" value="value-1">Option 1</vaadin-item>
        </vaadin-list-box>
      `,
      root,
    );
  };
</script>
```

web-padawan commented:
> This issue is actually reproducible with both Lit and Polymer.

Good catch, this seems to be caused by setting `selected` to true after attaching the item which is done to prevent resetting `selected` to `false` when setting select value to an item that is also `disabled`: https://github.com/vaadin/vaadin-select/pull/106

Note, we most probably don't need `selected` attribute on the clone since the original issue mentions style adjustments related to padding https://github.com/vaadin/vaadin-select/issues/103 - in V14 they are defined [here](https://github.com/vaadin/vaadin-select/blob/eefdfaae25484919bd2ada97192555a074ff866d/theme/lumo/vaadin-select-styles.html#L21-L25), but in V22 we use different selector:

https://github.com/vaadin/web-components/blob/6999a73838f3c851bfd2a5fe2b062b6eb003bd8a/packages/select/src/vaadin-select-value-button-styles.js#L21-L25

So IMO can remove `selected` from the item clone. I'd prefer to change that in a separate PR though.

tomivirkki commented on packages/item/src/vaadin-item-mixin.js:
```diff
@@ -38,6 +38,7 @@ export const ItemMixin = (superClass) =>
           value: false,
           reflectToAttribute: true,
           observer: '_selectedChanged',
+          sync: true,
```
nit: isn't this change unnecessary since running `performUpdate()` will sync the property change anyway?

vursen replied:
I guess it's still necessary because the first render is done asynchronously

web-padawan replied:
Right, seems unnecessary. Reverted.

web-padawan commented:
> Could both issues be resolved by ensuring the selected property is false before attaching the element? 

Makes sense, thanks. Updated and renamed the PR accordingly.

web-padawan commented:
UPD: unfortunately we still would need to remove `aria-selected` attribute with `selected` set to `false`.
